### PR TITLE
exercises: add armstrong-numbers

### PR DESCRIPTION
**Pull request by ee7**
To-do:

- [x] Handle possible (but untested) overflow in `sum`
- [x] Consider adding test cases to check correctness for inputs that overflow `sum`overflow handling
- [x] Support systems that aren't 64-bit: explicitly write `u64`, not `usize`.
- [x] Consider using `u128` as the input. The largest armstrong number requires 127 bits, so `u128` feels convenient, and would allow us to have a test for the largest armstrong number. But at least for an Exercism exercise, it might be better to stick with `u64` for now. Thoughts?

Things that we can leave for later:

- [ ] Consider adding a few more tests. For example, to check that wrapping arithmetic wasn't used, find an input number `n` whose armstrong sum is `n + 2^128`?
- [ ] Tweak exercise ordering in `config.json`.

### CODE REVIEW

File: exercises/practice/armstrong-numbers/armstrong_numbers.zig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub fn isArmstrongNumber(num: u64) bool {
+    @panic("please implement the isArmstrongNumber function");
+}
```
Comment (hiljusti) [inline exercises/practice/armstrong-numbers/armstrong_numbers.zig:3]: [Non-blocking] What do you think about also setting params to `_` sinks? Zig is a language with opinions so if you don't panic and don't use a variable you'll get errors. I think for the purposes of learning, we might want to add things like:

```
_ = num;
```

It's not strictly necessary for the code to compile, but it might be helpful

File: exercises/practice/armstrong-numbers/.meta/example.zig [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const std = @import("std");
+const math = std.math;
+
+/// Returns the number of digits in `n`.
+/// Uses only integer comparison (no repeated divison, nor log10).
```
Comment (hiljusti) [inline exercises/practice/armstrong-numbers/.meta/example.zig:5]: I'm curious why not log10. Especially if the hardware supports logarithm instructions, I think it might be more optimal (and [easier to quickly understand](https://probablydance.com/2021/07/29/a-logarithm-is-just-the-number-of-digits/)?) than multiplication

If I was mentoring, I'd personally also accept this, and maybe also prefer it:

```
return 1 + @floatToInt(u64, @trunc(@log10(@as(f64, num))));
```

What if this was in a comment as an alternative?

**Review by hiljusti — APPROVED** [review]
